### PR TITLE
bug #4590 - scrolling wallet send only after keyboard is open

### LINKED ISSUE #4590: Amount is overlapped by keyboard on "Send Transaction" screen (5.1" inches)
### Description

*Type*: Bug
*Summary*: when I tap on "Amount" I cannot see what I'm typing, because field is completely overlapped by keyboard.

#### Expected behavior
Amount is visible

#### Actual behavior
Amount is overlapped
amount1

### Reproduction

- Open Status
- Tap on Wallet
- Tap on "Send transaction"
- Type Amount

### Additional Information

* Status version: nightly 01/06/2018
* Operating System: iOS
* TF session (01:00 - 01:06): https://app.testfairy.com/projects/4803622-status/builds/8413233/sessions/165/?accessToken=s1KHv6cyuikO4xrRSOh/3zzsM78
* Device: Galaxy S7
* Screen: 5.1" inches

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8413233/sessions/165/?accessToken=s1KHv6cyuikO4xrRSOh/3zzsM78">TestFairy: Status 0.9.19 (6026) @165</a></strong></div></blockquote>

### PR DESCRIPTION
**Pull request by goranjovic**
fixes #4590

### Summary:

There was a race condition between focus on input field and keyboard opening, so it was common for wallet to scroll before the keyboard was open (i.e. not to scroll).

This PR fixes the race condition by making sure that screen is scrolled only after keyboard was open and only if the field is focused.

### Steps to test:
- Open Status on Android with smaller screen
- Wallet > Send and focus on amount field
- screen should scroll down so that amount field is still visible with the keyboard open

status: ready

### CODE REVIEW

**Review by dmitryn — APPROVED** [review]
@goranjovic have you considered calling `.scrollToEnd` on `keyboardDidShow` / `keyboardWillShow` events  instead of relying on timeouts?

**Review by flexsurfer — APPROVED** [review]

**Comment by goranjovic** [comment]
@dmitryn I haven't - that's a good idea

**Comment by goranjovic** [comment]
@dmitryn @flexsurfer Completely changed the PR, please re-review

File: src/status_im/ui/screens/wallet/send/views.cljs [hunk 1]
```diff
@@ -171,6 +172,21 @@
         [password-input-panel :t/signing-phrase-description in-progress?])
       (when in-progress? [react/view styles/processing-view])]]))
 
+;; MAIN SEND TRANSACTION VIEW
+(defn- send-transaction-view [{:keys [scroll] :as opts}]
+  (let [amount-input (atom nil)]
```
Comment (dmitryn) [inline src/status_im/ui/screens/wallet/send/views.cljs]: should be `reagent/atom`?

File: src/status_im/ui/screens/wallet/send/views.cljs [hunk 2]
```diff
@@ -171,6 +172,21 @@
...
+    (reagent/create-class
+     {:component-will-mount (fn [_]
+                              (.addListener react/keyboard
+                                            "keyboardDidShow"
```
Comment (dmitryn) [inline src/status_im/ui/screens/wallet/send/views.cljs]: This is Android only. For iOS, there is separate event `keyboardWillShow` https://github.com/status-im/status-react/blob/develop/src/status_im/ios/core.cljs#L25

**Review by dmitryn — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

**Comment by lukaszfryc** [comment]
@goranjovic the screen scrolls fine. But, I can't open the advanced settings. Clicking on the Advanced button does not open the section.

**Comment by goranjovic** [comment]
@lukaszfryc huh, how did that happen.. will look into

**Comment by dmitryn** [comment]
@lukaszfryc issue with clicking "Advanced" has been fixed, please take a look.

**Pull request merged**